pkg/destinations/destinations.go

```diff
@@ -83,24 +87,76 @@ func (m *DestinationManager) TestCredentials(creds config.Destination) error {
 	return nil
 }
 
-func (m *DestinationManager) Destination(ctx context.Context, databaseID int64) (Destination, error) {
+func (m *DestinationManager) UpdateDestination(ctx context.Context, dest dmodels.Destination, creds map[string]any) error {
+	err := m.storage.Database.UpdateDestination(ctx, dest)
+	if err != nil {
+		return err
+	}
+
+	ds, err := json.Marshal(creds)
+	if err != nil {
+		return err
+	}
+
+	cn := credentialName(dest.TeamID, dest.ID, dest.Name)
+	err = m.storage.Vault.SetCredential(cn, string(ds))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *DestinationManager) CreateDestination(ctx context.Context, teamID uint, dest config.Destination) (uint, error) {
+	// TODO breadchris create a destination in the database and store the credentials in the vault
+	d, err := m.storage.Database.CreateDestination(ctx, teamID, dest.Name, dest.Type)
+	if err != nil {
+		return 0, err
+	}
 
+	ds, err := json.Marshal(dest.Settings)
+	if err != nil {
+		return 0, err
+	}
+
+	cn := credentialName(teamID, d.ID, dest.Name)
+	err = m.storage.Vault.SetCredential(cn, string(ds))
+	if err != nil {
+		return 0, err
+	}
+	return d.ID, nil
+}
+
+func (m *DestinationManager) GetDestination(ctx context.Context, databaseID uint) (Destination, error) {
 	if m.mux.TryLock(databaseID) {
 		defer m.mux.Unlock(databaseID)
 
-		var dest Destination
+		var (
+			dest     Destination
+			settings map[string]any
+		)
 
 		dest, ok := m.pool[databaseID]
 		if ok {
 			return dest, nil
 		}
 
+		// XXX breadchris does the caller verify existence and authz for this database?
+		// is this call needed?
 		creds, err := m.storage.Database.GetDestinationCredentials(ctx, databaseID)
 		if err != nil {
 			return nil, err
 		}
 
-		settings := creds.Settings.Data()
+		cn := credentialName(creds.TeamID, databaseID, creds.Name)
```

Comment:
Shall we store the credential name in the database in the Destination model? This way when we inevitably change the naming scheme we won't lose anything. Also useful to track down orphaned items in the vault.